zenodo/modules/openaire/tasks.py

```diff
@@ -134,5 +143,30 @@ def openaire_delete(record_uuid=None, original_id=None, datasource_id=None):
 
         if not res.ok or (res_beta and not res_beta.ok):
             raise OpenAIRERequestError(res.text)
+
+        # Remove from failures cache
+        current_cache.delete(
+            'openaire_direct_index:{}'.format(record['recid']))
```

Comment:
lack of knowledge question: how/when are the failed tasks registered under this key in the cache?

Reply:
Nice catch, you're right, failed "deletions" are not registered... [Direct indexing sets failed recids](https://github.com/zenodo/zenodo/blob/0059ad30e73b229de5b8394f5f73b4020dbf7b11/zenodo/modules/openaire/tasks.py#L99) though. I will add it for the deletions as well.